### DISCUSSION TITLE
Custom Wallet do something when i click custom wallet

### DISCUSSION THREAD

**Discussion by FurkanSezal** [rainbow-me/rainbowkit#1427] [Q&A] [answered]
Hi, 
I build a custom wallet. I want to do when the user click to connect in the modal, i want to check if the wallet injected or not. if its not injected, i want to redirect the user to download page. how can i do that?

**Comment by DanielSinclair** [comment]
Here is an example from the Rainbow Wallet connector [implementation](https://github.com/rainbow-me/rainbowkit/blob/a8690ef3a374af3fd00318e72fe79306bf089071/packages/rainbowkit/src/wallets/walletConnectors/rainbowWallet/rainbowWallet.ts).
```ts
function isRainbow(ethereum: NonNullable<typeof window['ethereum']>) {
  // `isRainbow` needs to be added to the wagmi `Ethereum` object
  const isRainbow = Boolean(ethereum.isRainbow);
  if (!isRainbow) return false;
  return true;
}

const isRainbowInjected =
  typeof window !== 'undefined' &&
  typeof window.ethereum !== 'undefined' &&
  isRainbow(window.ethereum);
```

You'll then want to return the `installed` param in the prepared connector. `shouldUseWalletConnect` in this case is `true` when `isRainbowInjected` is `false`, as RainbowKit only supports 1 type of connector at a time (walletconnect qr code or injected wallet), and the typical expected behavior is to prefer browser extensions.
```ts
installed: !shouldUseWalletConnect ? isRainbowInjected : undefined,
```

**Reply by FurkanSezal** [reply]
lets say the user doent have installed rainbow wallet. The user click the rainbow wallet option. Because the user doent have rainbow wallet, i want to redirect the user to a download page. if only clicks the rainbow wallet. How can i do it?

**Reply by DanielSinclair** [reply, marked as answer]
@FurkanSezal We don't current expose a callback if a user chooses a particular wallet that is not installed, but we do show very clear UI that assists users in downloading extensions and mobile/desktop apps in that fallback scenario. We account for different browsers and platforms in this flow, so it is recommended that your dApp relies on this instead of rolling your own.
<img width="763" alt="Screenshot 2023-08-02 at 1 23 08 AM" src="https://github.com/rainbow-me/rainbowkit/assets/4412473/7d6cd4da-f13e-405e-89b1-5fe7b12f485f">
